EIPS/eip-9999.md

```diff
@@ -0,0 +1,447 @@
+4. **Timing**: The CL knows whether chunks arrived timely
+
+### Why Separate CALs from Chunks?
+
+Separating Chunk Access Lists (containing state diffs) from chunks provides several benefits:
+
+1. **Independent Propagation**: CALs are typically much smaller than chunks (containing only state diffs) and propagate faster
+2. **Parallel Execution**: Multiple chunks can execute independently by applying prior CALs to reconstruct required pre-states
+3. **Modularity**: State diffs are logically separate from transaction data
+4. **Future Compatibility**: CALs can be extended with additional metadata without modifying chunk structure
+5. **Streaming Validation**: Validators can begin executing chunks as soon as required CALs arrive
+
+### Why Semantic Chunking Over Simple Fragmentation?
+
+Unlike simple byte-level fragmentation, semantic chunking:
+
+1. **Enables Streaming Validation**: Chunks are self-contained execution units that can be validated as they arrive
+2. **Bounds Resources**: Each chunk has maximum gas limits, bounding memory and CPU usage
+
+### Choice of Parameters
+
+- **`CHUNK_GAS_LIMIT = 2**24`** (16.7M gas): Balances parallelization benefits with chunk overhead. Large enough for complex transactions, small enough for bounded proving
+- **`MAX_CHUNKS_PER_BLOCK = 256`**: Supports up to 268 Mgas blocks
+- **`MAX_TRANSACTIONS_PER_CHUNK = 2**10`** (1,024): Smallest power of two that is higher than the number of transactions in a chunk full of basic ETH transfers (21,000 gas per tx)
+
+## Backwards Compatibility
+
+This EIP introduces breaking changes to:
+
+1. **Block Structure**: Replaces monolithic execution payload with chunked structure
+2. **Network Protocol**: Adds new gossip topics for chunks and CALs
+3. **Engine API**: Requires new validation methods
+4. **Fork Choice**: Adds three-phase validation logic
+
+These changes require a coordinated hard fork.
+
+## Security Considerations
+
+### Data Availability
+
+Chunk separation creates new data availability requirements:
+
+1. **Withholding Attacks**: Proposers could withhold specific chunks or CALs to make blocks unvalidatable
+   - **Mitigation**: Attestation rules require full chunk availability before voting
+   - **Detection**: Missing chunks are easily identified by commitment mismatches
+
+2. **Partial Withholding**: Withholding CALs (state diffs) while publishing chunks
+   - **Impact**: Delays streaming execution but doesn't prevent sequential validation
+   - **Mitigation**: CALs can be reconstructed by executing chunks sequentially to capture state diffs
+
+3. **Orphan Chunk Handling**: Chunks arriving before beacon blocks
+   - **Storage**: Orphan chunks stored temporarily with (slot, proposer, index) identification
+   - **Validation**: Confirmed against commitments when beacon block arrives
+   - **Limits**: Bounded orphan storage prevents DoS attacks
+
```

Comment:
There is also the "attack" to stream chunks + CALs in the wrong order, i.e. start with the last one and then stream to the start. Then it is not possible to start executing before the final chunk/CAL has been received.